cpp/Integri/XsdComparator/XsdComparator.cpp

```diff
@@ -0,0 +1,219 @@
+#include "XsdComparator.h"
+#include <iostream>
+#include <stdio.h>
+#include <tchar.h>
+#include <windows.h>
+#include <atlbase.h>
+#include <comutil.h>
+
+using namespace std;
+
+XsdComparator::XsdComparator(Reporter* reporter)
+{
+	this->reporter = reporter;
+}
+
+void XsdComparator::Compare(const std::string file1, const std::string file2)
+{
+	try
+	{
+		MSXML2::IXMLDOMElement* firstRoot;
```

Comment:
Always initialize local variables (even if it is to nullptr) to avoid undefined behavior when working with uninitialized variables. Notice, that the compiler would "carve" a piece of stack memory for those and thus the memory might be all 0 (in a good case, which is essentially an equivalent to a nullptr) or worse it might be some random garbage, which doesn't compare to nullptr and you might be getting an ungly surprise sometimes in the future :-)